File: rtx.php
<?php



function rtx_post($url, $post_data) {  
  
  $postdata = http_build_query($post_data);  
  $options = array(  
    'http' => array(  
      'method' => 'POST',  
      'header' => 'Content-type:application/x-www-form-urlencoded',  
      'content' => $postdata,  
      'timeout' => 15 * 60 
    )  
  );  
  $context = stream_context_create($options);  
  $result = file_get_contents($url, false, $context);  
  
  return $result;  
}  
  
$msg =  iconv('UTF-8', 'gbk', $_GET['msg']);


$post_data = array(  
  'sender' => 'robot',  
  'pwd' => 'robot',  
  'receivers' => 'julian',  
  'msg' => $msg,  
  'sessionid' => '123123'  
);  

$rep = rtx_post('http://192.168.172.2:8012/SendIM.cgi', $post_data);  

echo $rep;

File: rtx.py
#!/usr/bin/env python
# -*- coding: gbk -*-
# -*- coding: utf_8 -*-
import urllib, httplib

httpClient = None
helloMsg = 'Hello Developer! \n\nAbby 的产品需求更新了 \n\n赶快去瞧瞧\n\nhttp://github.panli.com/abby/PanliByAbby/commits/master'
robotM = '\n\n  我是来自运维团队的R2机器人的消息通知'
devs = 'julian;jom;lewis;manu;jessica;yoko;wilson;abby;jean'
sessionid = 'devsAllIM'

try:
    params = urllib.urlencode({'sender': 'robot', 'pwd': 'robot', 'receivers': 'julian', 'msg': helloMsg+robotM, 'sessionid': sessionid})
    headers = {'Content-type': 'application/x-www-form-urlencoded', 'Accept': 'text/plain'}
    httpClient = httplib.HTTPConnection('172.20.7.29', 8012, timeout=10)
    httpClient.request('POST', '/SendIM.cgi', params, headers)
    response = httpClient.getresponse()
    print response.status
    print response.reason
    print response.read()
    print response.getheaders()
except Exception, e:
    print e
finally:
    if httpClient:
        httpClient.close()